ocaml/forkexecd/lib/forkhelpers.ml

```diff
@@ -203,10 +213,23 @@ let safe_close_and_exec ?env stdin stdout stderr
 
       let s =
         match response with
-        | Fe.Setup_response s ->
+        | Ok (Fe.Setup_response s) ->
             s
-        | _ ->
-            failwith "Failed to communicate with forking executioner"
+        | Ok status ->
+            let msg =
+              Printf.sprintf
+                "%s: Received unexpected reply from forking executioner (%s)"
+                __FUNCTION__
+                (Fe.ferpc_to_string status)
+            in
+            failwith msg
+        | Error err ->
+            let msg =
+              Printf.sprintf
+                "%s: Received invalid reply from forking executioner (%s)"
+                __FUNCTION__ err
+            in
+            failwith msg
```

Comment:
This is fine, you can also use `Fmt.failwith` instead which takes a printf format string and does the internal printf + failwith internally.